File: src/Book.cpp
//
//  Book.cpp
//  BookStore
//
//  Created by Dan Jiang on 2018/5/17.
//  Copyright © 2018年 Dan Jiang. All rights reserved.
//

#include "Book.hpp"

using namespace Store;

std::string Book::toJson() const {
    json11::Json json = json11::Json::object {
        { "title", title },
        { "price", price },
        { "author", author },
        { "pages", pages }
    };
    return json.dump();
}

std::string Book::toString() const {
    std::stringstream os;
    
    os << "Book(" << this->title << ", " << this->price << ", " << this->author << ", " << this->pages << ")";

    return os.str();
}

File: ios_project/BookStore/BookStoreWrapper/Book.m
//
//  Book.m
//  BookStoreWrapper
//
//  Created by Dan Jiang on 2018/5/17.
//  Copyright © 2018年 Dan Jiang. All rights reserved.
//

#import "Book.h"

@implementation Book

- (instancetype)initWithDict:(NSDictionary *)dict {
    self = [super init];
    if (self) {
        self.title = dict[@"title"];
        self.price = ((NSNumber *)dict[@"price"]).doubleValue;
        self.author = dict[@"author"];
        self.pages = ((NSNumber *)dict[@"pages"]).intValue;
    }
    return self;
}

@end
